### PR TITLE
Ruby code

### PR DESCRIPTION
**Pull request by melmothx**
This is an ongoing work, but this branch should be in good shape for merging anytime.

Thanks.

### CODE REVIEW

**Comment by melmothx** [comment]
Output is here: https://xapian.amusewiki.org/ruby/

File: code/ruby/index_ranges.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'xapian'
+require 'json'
+require_relative 'support'
+
+def index(data_path, db_path)
+  # puts "#{data_path} #{db_path}"
```
Comment (ojwb) [inline code/ruby/index_ranges.rb:9]: I assume this (and similar) are leftovers from debugging so I've pushed a commit to remove them.

File: code/ruby/index1.rb [hunk 2]
```diff
@@ -1,36 +1,32 @@
...
     doc.add_boolean_term(idterm)
     db.replace_document(idterm, doc)
   end
 end
 ### End of example code.
 
-if ARGV.length < 2
-  abort "Usage #{__FILE__} DATAPATH DBPATH"
-end
+abort "Usage #{__FILE__} DATAPATH DBPATH" if ARGV.length < 2
```
Comment (ojwb) [inline code/ruby/index1.rb:30]: I find it curious that a style tool should push towards this as it seems less clear in a case like this where the action is the exceptional case as it emphasises the action over the check, but here the `ARGV.length < 2` is the part I'd want the reader to focus on (`<action> if <condition>` seems reasonable when the action is the interesting part).

(To be clear, this is not a review comment saying to change this - really just a thought I had while skimming the changes.)

**Comment by ojwb** [comment]
Merged, many thanks.

**Pull request merged**